climbing_stairs/climbing_stairs.py

```diff
@@ -1,9 +1,21 @@
 #!/usr/bin/python
 
 import sys
-
+n = 50
 def climbing_stairs(n, cache=None):
-  pass 
+  if n == 1 or n == 0:
+    return 1
+  elif n == 2:
+    return 2
+  elif n == 3:
+    return 4
+  elif cache and cache[n] > 0:
```

Comment:
nice most people don't think about dealing with the option of inputting a cache or not.